src/test/shell/bazel/runfiles_test.sh

```diff
@@ -261,4 +261,154 @@ EOF
     bazel build --spawn_strategy=local --nobuild_runfile_links --enable_runfiles=false //:out
 }
 
+function setup_runfiles_tree_file_type_changes {
+  mkdir -p rules
+  touch rules/BUILD
+  cat > rules/defs.bzl <<'EOF'
+def _make_fake_executable(ctx):
+    fake_executable = ctx.actions.declare_file(ctx.label.name)
+    ctx.actions.write(
+        output = fake_executable,
+        content = "echo 'i do nothing'",
+        is_executable = True,
+    )
+    return fake_executable
+
+def _tree_artifact(ctx):
+    d = ctx.actions.declare_directory("lib")
+    args = ctx.actions.args()
+    args.add(d.path)
+    ctx.actions.run_shell(
+        outputs = [d],
+        inputs = [],
+        arguments = [args],
```

Comment:
I'd just add `arguments = [d.path]` here and avoid the `args` to make this easier to read.